### PR TITLE
Feat: Create notification service

### PR DESCRIPTION
**Pull request by ZixMai**
1. Created grpc base for notification service
2. Created db handler for notification service
3. Created notification repository for notification service
4. Created grpc requests methods for notification service
5. Created interceptor for notification service
6. Created proto file for notification service
7. Fixed some issues in event service and dockerfiles

### CODE REVIEW

**Review by MaxPlays35 — APPROVED** [review]

File: notification_service/repository/mock_notification_repository.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        """
+        try:
+            return next(
+                notification
+                for notification in self._notifications
+                if notification.id == notification_id
+                and notification.deleted_at is None
+            )
+        except StopIteration:
+            raise ValueNotFoundError("Notifications not found")
```
Comment (Nhsdkk) [inline notification_service/repository/mock_notification_repository.py]: Should be "Notification not found"

File: notification_service/repository/mock_notification_repository.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+from typing import List
+from uuid import uuid4
+
+from errors.value_not_found_error import ValueNotFoundError
+from src.models.notification import Notification
+
+from repository.notification_repository_interface import NotificationRepositoryInterface
+
+
+class MockNotificationRepositoryImpl(NotificationRepositoryInterface):
```
Comment (Nhsdkk) [inline notification_service/repository/mock_notification_repository.py:14]: Annotate with singleton

File: notification_service/repository/notification_repository_impl.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        ValueNotFoundError
+            No notifications were found for given notifications ids.
+
+        """
+        db_notifications: Optional[
+            List[PrismaNotification]
+        ] = await self._db_client.db.notification.find_many(
+            where={
+                "id": {
+                    "in": [notification_id for notification_id in notifications_ids]
```
Comment (Nhsdkk) [inline notification_service/repository/notification_repository_impl.py]: should be just "in": notifications_ids

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

File: notification_service/src/notification_service_impl.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        Function that need to be bind to the server that returns notifications list.
+    async get_all_notifications(request, context)
+        Function that need to be bind to the server that returns all notifications in list.
+    async create_notification(request, context)
+        Function that need to be bind to the server that creates the notification.
+    async update_notification(request, context)
+        Function that need to be bind to the server that updates the notification.
+    async delete_notification(request, context)
+        Function that need to be bind to the server that deletes the notification.
+    """
```
Comment (Nhsdkk) [inline notification_service/src/notification_service_impl.py:42]: add new line

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Pull request merged**